include/cloysterhpc/services/repo.h

```diff
@@ -47,4 +50,60 @@ class repo {
     static void load_repository(std::filesystem::path path);
 };
 
+class repository_exception : public std::runtime_error {
+public:
+    explicit repository_exception(const std::string& msg)
+        : std::runtime_error(msg)
+    {
+    }
+};
```

Comment:
@viniciusferrao opinions on this? if we'll not be handling exceptions is there any point in creating our own exceptions?

Reply:
This is just a WIP